shared/Utils.js

```diff
@@ -3,5 +3,50 @@
 export default {
 	stripHtmlTags (inputString) {
 		return inputString.replace(/<\/?[^>]+(>|$)/g, "");
+	},
+
+	/**
+	 * Generate the navigation menu based on the response from the Wordpress API.
+	 * The retun menu array will:
+	 * 1. Exclude all pages whose menu_order === 0;
+	 * 2. Append "News" and "Views" menu items to the end.
+	 * @param {Array<Object>} sitePages - All menu pages sorted by menu_order.
+	 * @param {Boolean} includeHomeItem - A flag indicating if the home menu item should be included.
+	 * @return {Array<Object>} An array of object including the slug, menu title and href.
+	 */
+	generateNavMenu (sitePages, includeHomeItem) {
+		// Don't include the menu item for "home" by default
+		includeHomeItem = includeHomeItem === undefined ? false : includeHomeItem;
+		const menuItems = [];
+		const homeMenuItem = [];
+		const itemsToAppend = [{
+			slug: "news",
+			title: "News",
+			href: "/news"
+		}, {
+			slug: "views",
+			title: "Views",
+			href: "/views"
+		}];
+
+		sitePages.forEach((oneSitePage) => {
+			if (includeHomeItem && oneSitePage.slug === "home") {
+				homeMenuItem.push({
+					slug: oneSitePage.slug,
+					title: "Home",
+					href: "/"
+				});
+			}
```

Comment:
I wonder if you can simplify things here by using [unshift](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift).

It seems that `oneSitePage.slug === "home"` will happen at the same time as `oneSitePage.menu_order < 0` otherwise you'd run the risk of adding the home page twice, and I suppose broken in the second case. Instead of pushing to `homeMenuItem` you just unshift to `menuItems` which would put it at the front of the list.

Reply:
Good catch. To avoid adding the home page twice, I will modify the condition of this block to `includeHomeItem && oneSitePage.slug === "home" && oneSitePage.menu_order === 0` so that the home page will only be added to `homeMenuItem` when it isn't included in the nav menu. Please let me know if you notice a flaw with this solution. Thanks.

Reply:
Do you need to check both `menu_order` and `slug`. Will home always have a `menu_order` that is 0?

Reply:
Based on the current design that excludes the home page from the navigation menu on the header, `menu_order` for the home will always be 0, unless there's a design change.